### DISCUSSION TITLE
What's the easiest way to go from pixbuf to Pixmap?

### DISCUSSION THREAD

**Discussion by foranuj** [pymupdf/PyMuPDF-Utilities#100] [General]
Would something like this be the answer?

```
def pixbuf_to_pixmap(pixbuf):
    width = pixbuf.get_width()
    height = pixbuf.get_height()
    pixels = pixbuf.get_pixels()

    return fitz.Pixmap(fitz.csRGB, width, height, bytearray(pixels), False)
```

**Comment by JorjMcKie** [comment]
The overloaded constructor you are referring to is correct.
You can use any object type of `bytes`, `bytearray` of `io.BytesIO`. In so far, a prior conversion to bytearray is not necessary if `pixbuf.get_pixels()` already is `bytes`.
This area must however be structured like a `Pixmap.samples`.

**Reply by foranuj** [reply]
You think this or a cleaner version of this code should be part of the library to allow easy conversion between pix buf and pixmap?
Currently there doesn't seem to be a function available to use.

**Reply by JorjMcKie** [reply]
what is pixbuf actually?

**Reply by JorjMcKie** [reply]
If just being given a bytes-like memory area, it could be the `samples` of 3 or 4 different pixmaps. You need all the other parameters to interpret it correctly. So there for me there seems to be no easier / more elegant constructor, if we don't want to talk about sheer mannerisms.
But I may be wrong.

**Reply by foranuj** [reply]
[pixbuf -> GdkPixbuf 
](https://docs.gtk.org/gdk-pixbuf/class.Pixbuf.html)
